components/automate-cli/cmd/chef-automate/config.go

```diff
@@ -119,47 +137,162 @@ func runShowCmd(cmd *cobra.Command, args []string) error {
-		input, err := ioutil.ReadFile(args[0])
-		if err != nil {
-			return nil
+		if configCmdFlags.postgresql {
+			for i := 0; i < 1; i++ {
+				executePatchOnRemote(sshUser, sshPort, sskKeyFile, infra.Outputs.PostgresqlPrivateIps.Value[i], args[0], "pg")
+			}
+		}
+		if configCmdFlags.opensearch {
+			for i := 0; i < 1; i++ {
+				executePatchOnRemote(sshUser, sshPort, sskKeyFile, infra.Outputs.OpensearchPrivateIps.Value[i], args[0], "os")
+			}
 		}
-		err = ioutil.WriteFile(AUTOMATE_HA_AUTOMATE_CONFIG_FILE, input, 0644)
+	} else {
+
+		cfg, err := dc.LoadUserOverrideConfigFile(args[0])
 		if err != nil {
-			writer.Printf("error in patching automate config to automate HA")
+			return status.Annotate(err, status.ConfigError)
+		}
+		if err = client.PatchAutomateConfig(configCmdFlags.timeout, cfg, writer); err != nil {
 			return err
 		}
-		return executeDeployment(args)
-	}
-
-	if err = client.PatchAutomateConfig(configCmdFlags.timeout, cfg, writer); err != nil {
-		return err
 	}
 
 	writer.Success("Configuration patched")
 	return nil
 }
 
+func executePatchOnRemote(sshUser string, sshPort string, sshKeyFile string, ip string, path string, remoteType string) {
+
+	pemBytes, err := ioutil.ReadFile(sshKeyFile)
+	if err != nil {
+		writer.Errorf("Unable to read private key: %v", err)
+	}
+	signer, err := ssh.ParsePrivateKey(pemBytes)
+	if err != nil {
+		writer.Errorf("Parsing key failed: %v", err)
+	}
+	var (
+		keyErr *knownhosts.KeyError
+	)
+
+	config := &ssh.ClientConfig{
+		User: sshUser,
+		Auth: []ssh.AuthMethod{ssh.PublicKeys(signer)},
+		HostKeyCallback: ssh.HostKeyCallback(func(host string, remote net.Addr, pubKey ssh.PublicKey) error {
+			kh := checkKnownHosts()
+			hErr := kh(host, remote, pubKey)
+			// Reference: https://blog.golang.org/go1.13-errors
+			// To understand what errors.As is.
+			if errors.As(hErr, &keyErr) && len(keyErr.Want) > 0 {
+				// Reference: https://www.godoc.org/golang.org/x/crypto/ssh/knownhosts#KeyError
+				// if keyErr.Want slice is empty then host is unknown, if keyErr.Want is not empty
+				// and if host is known then there is key mismatch the connection is then rejected.
+				writer.Printf("WARNING: Given hostkeystring is not a key of %s, either a MiTM attack or %s has reconfigured the host pub key.", host, host)
+				return keyErr
+			} else if errors.As(hErr, &keyErr) && len(keyErr.Want) == 0 {
+				// host key not found in known_hosts then give a warning and continue to connect.
+				writer.Printf("WARNING: %s is not trusted, adding this key to known_hosts file.", host)
+				return addHostKey(host, remote, pubKey)
+			}
+			writer.Printf("Pub key exists for %s.", host)
+			return nil
+		}),
+	}
+
+	// Open connection
+	conn, err := ssh.Dial("tcp", ip+":"+sshPort, config)
+	if conn == nil || err != nil {
+		writer.Errorf("dial failed:%v", err)
+		return
+	}
+	defer conn.Close()
+
+	// Open session
+	session, err := conn.NewSession()
+	if err != nil {
+		writer.Errorf("session failed:%v", err)
```

Comment:
what we are resuming here when we get an error

Reply:
Added error return